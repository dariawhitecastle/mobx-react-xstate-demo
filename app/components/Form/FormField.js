/* eslint-disable react/prop-types */
/**
 * Form
 */

import React from 'react'
import { StyledFormField } from './FormField.styles'
import { doc } from './Form.doc'
import helpers from '../../utils/helpers'

const createFormField = props => {
  const {
    component,
    error,
    help,
    htmlFor,
    label,
    name,
    pad,
    required,
    validate,
    children,
    value,
    ...rest
  } = props
  return (
    <StyledFormField
      component={component}
      error={error}
      help={help}
      htmlFor={htmlFor}
      label={label}
      name={name}
      pad={pad}
      required={required}
      validate={validate}
      {...value}
      {...rest}
    >
      {children}
    </StyledFormField>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createFormField,
})
