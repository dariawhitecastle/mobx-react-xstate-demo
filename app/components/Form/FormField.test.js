import React from 'react'
import renderer from 'react-test-renderer'
import styled from 'styled-components'
import 'jest-styled-components'

import { Grommet, TextInput } from 'grommet'
import FormField from './FormField'

const CustomFormField = styled(FormField)`
  font-size: 40px;
`

test('renders', () => {
  const component = renderer.create(
    <Grommet>
      <FormField />
      <FormField>
        <TextInput />
      </FormField>
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders label', () => {
  const component = renderer.create(
    <Grommet>
      <FormField label="test label" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders help', () => {
  const component = renderer.create(
    <Grommet>
      <FormField help="test help" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders error', () => {
  const component = renderer.create(
    <Grommet>
      <FormField error="test error" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders htmlFor', () => {
  const component = renderer.create(
    <Grommet>
      <FormField htmlFor="test-id" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders custom formfield', () => {
  const component = renderer.create(
    <Grommet>
      <CustomFormField htmlFor="test-id" />
    </Grommet>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
