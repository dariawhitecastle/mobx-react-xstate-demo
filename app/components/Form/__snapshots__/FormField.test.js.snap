// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c5 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
  border: none;
}

.c5::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c5::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c5::-moz-placeholder {
  color: #AAAAAA;
}

.c5:-ms-input-placeholder {
  color: #AAAAAA;
}

.c5::-moz-focus-inner {
  border: none;
  outline: none;
}

.c4 {
  position: relative;
  width: 100%;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 input {
  height: 60px;
  padding-left: 20px;
  font-size: 16px;
  font-weight: normal;
  color: rgb(68,68,68);
}

.c1 label + div {
  border: solid 1px;
  border-radius: 4px;
  border-color: rgb(170,170,170);
}

.c1 label + div:focus-within {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 div {
  border-bottom: none;
}

.c1 label {
  margin-left: 0;
}

.c1 label div input {
  display: none;
}

.c1 label div div {
  margin-top: 20px;
  border: solid 1px rgb(68,68,68);
}

.c1 label div div svg {
  stroke: rgb(68,68,68);
}

.c1 label span {
  padding-top: 20px;
  font-size: 16px;
  color: rgb(68,68,68);
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <div
      className="c3"
    />
  </div>
  <div
    className="c1 c2"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      >
        <input
          autoComplete="off"
          className="c5"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders custom formfield 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 input {
  height: 60px;
  padding-left: 20px;
  font-size: 16px;
  font-weight: normal;
  color: rgb(68,68,68);
}

.c2 label + div {
  border: solid 1px;
  border-radius: 4px;
  border-color: rgb(170,170,170);
}

.c2 label + div:focus-within {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c2 div {
  border-bottom: none;
}

.c2 label {
  margin-left: 0;
}

.c2 label div input {
  display: none;
}

.c2 label div div {
  margin-top: 20px;
  border: solid 1px rgb(68,68,68);
}

.c2 label div div svg {
  stroke: rgb(68,68,68);
}

.c2 label span {
  padding-top: 20px;
  font-size: 16px;
  color: rgb(68,68,68);
}

.c1 {
  font-size: 40px;
}

@media only screen and (max-width:768px) {
  .c3 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2 c3"
  >
    <div
      className="c4"
    />
  </div>
</div>
`;

exports[`renders error 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px #FF4040;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c4 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
  color: #FF4040;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 input {
  height: 60px;
  padding-left: 20px;
  font-size: 16px;
  font-weight: normal;
  color: rgb(68,68,68);
}

.c1 label + div {
  border: solid 1px;
  border-radius: 4px;
  border-color: rgba(255,64,64,1);
}

.c1 label + div:focus-within {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 div {
  border-bottom: none;
}

.c1 label {
  margin-left: 0;
}

.c1 label div input {
  display: none;
}

.c1 label div div {
  margin-top: 20px;
  border: solid 1px rgb(68,68,68);
}

.c1 label div div svg {
  stroke: rgb(68,68,68);
}

.c1 label span {
  padding-top: 20px;
  font-size: 16px;
  color: rgb(68,68,68);
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-bottom: solid 1px #FF4040;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <div
      className="c3"
    />
    <span
      className="c4"
    >
      test error
    </span>
  </div>
</div>
`;

exports[`renders help 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c3 {
  margin-left: 12px;
  font-size: 18px;
  line-height: 24px;
  color: #777777;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 input {
  height: 60px;
  padding-left: 20px;
  font-size: 16px;
  font-weight: normal;
  color: rgb(68,68,68);
}

.c1 label + div {
  border: solid 1px;
  border-radius: 4px;
  border-color: rgb(170,170,170);
}

.c1 label + div:focus-within {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 div {
  border-bottom: none;
}

.c1 label {
  margin-left: 0;
}

.c1 label div input {
  display: none;
}

.c1 label div div {
  margin-top: 20px;
  border: solid 1px rgb(68,68,68);
}

.c1 label div div svg {
  stroke: rgb(68,68,68);
}

.c1 label span {
  padding-top: 20px;
  font-size: 16px;
  color: rgb(68,68,68);
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <span
      className="c3"
    >
      test help
    </span>
    <div
      className="c4"
    />
  </div>
</div>
`;

exports[`renders htmlFor 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 input {
  height: 60px;
  padding-left: 20px;
  font-size: 16px;
  font-weight: normal;
  color: rgb(68,68,68);
}

.c1 label + div {
  border: solid 1px;
  border-radius: 4px;
  border-color: rgb(170,170,170);
}

.c1 label + div:focus-within {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 div {
  border-bottom: none;
}

.c1 label {
  margin-left: 0;
}

.c1 label div input {
  display: none;
}

.c1 label div div {
  margin-top: 20px;
  border: solid 1px rgb(68,68,68);
}

.c1 label div div svg {
  stroke: rgb(68,68,68);
}

.c1 label span {
  padding-top: 20px;
  font-size: 16px;
  color: rgb(68,68,68);
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <div
      className="c3"
    />
  </div>
</div>
`;

exports[`renders label 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c3 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 input {
  height: 60px;
  padding-left: 20px;
  font-size: 16px;
  font-weight: normal;
  color: rgb(68,68,68);
}

.c1 label + div {
  border: solid 1px;
  border-radius: 4px;
  border-color: rgb(170,170,170);
}

.c1 label + div:focus-within {
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c1 div {
  border-bottom: none;
}

.c1 label {
  margin-left: 0;
}

.c1 label div input {
  display: none;
}

.c1 label div div {
  margin-top: 20px;
  border: solid 1px rgb(68,68,68);
}

.c1 label div div svg {
  stroke: rgb(68,68,68);
}

.c1 label span {
  padding-top: 20px;
  font-size: 16px;
  color: rgb(68,68,68);
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    padding: 0px;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <label
      className="c3"
    >
      test label
    </label>
    <div
      className="c4"
    />
  </div>
</div>
`;
